server/admin_management/tests/services/test_user_validator.py

```diff
@@ -0,0 +1,178 @@
+import logging
+import pytest
+from fastapi import HTTPException
+from mock import patch
+
+from api.app.services.validator.user_validator import UserValidator
+from api.app.integration.idim_proxy import IdimProxyService
+from api.app.schemas import Requester, TargetUser
+from api.app.constants import ERROR_CODE_INVALID_REQUEST_PARAMETER, UserType
+from tests.constants import (
+    TEST_USER_NAME_IDIR,
+    TEST_USER_GUID_IDIR,
+    TEST_USER_NAME_BCEID,
+    TEST_USER_GUID_BCEID,
+    TEST_USER_BUSINESS_GUID_BCEID,
+)
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+TEST_NEW_TARGET_USER_IDIR = {
+    "user_type_code": UserType.IDIR,
+    "user_name": TEST_USER_NAME_IDIR,
+    "user_guid": TEST_USER_GUID_IDIR,
+}
+
+TEST_NEW_TARGET_USER_BCEID_LOAD_2 = {
+    "user_type_code": UserType.BCEID,
+    "user_name": TEST_USER_NAME_BCEID,
+    "user_guid": TEST_USER_GUID_BCEID,
+}
+
+TEST_REQUESTER = {
+    "user_id": 11,  # this is a faked id
+    "user_type_code": UserType.IDIR,
+    "user_name": TEST_USER_NAME_IDIR,
+    "user_guid": TEST_USER_GUID_IDIR,
+}
+
+MOCK_SERACH_IDIR_RETURN = {
+    "found": True,
+    "guid": TEST_NEW_TARGET_USER_IDIR.get("user_guid"),
+    "userId": TEST_NEW_TARGET_USER_IDIR.get("user_name"),
+}
+
+MOCK_SERACH_BCEID_RETURN = {
+    "found": True,
+    "guid": TEST_NEW_TARGET_USER_BCEID_LOAD_2.get("user_guid"),
+    "userId": TEST_NEW_TARGET_USER_BCEID_LOAD_2.get("user_name"),
+    "businessGuid": TEST_USER_BUSINESS_GUID_BCEID,
+}
+
+
+class TestUserValidatorClass(object):
+    """
+    Testing UserValidator class with mocked IDIM API calls.
+    """
+
+    def setup_class(self):
+        # local valid mock requester
+        self.requester_idir = Requester(**TEST_REQUESTER)
+
+    @patch.object(IdimProxyService, "search_idir")
```

Comment:
Only wondering. Do you know if this "patch" to mock the service need to be reset before each test case begins?
Also, do you know if we can verify if the mocked service is being called?
Thanks!

Reply:
I'm actually not sure, but if you check the `server/backend/testspg/crud/test_UserRoleValidator.py` file, we used patch before the mock forest client api return. And we can verify if the service is being called
<img width="1277" alt="Screen Shot 2024-06-04 at 2 45 06 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/3e9996a6-a991-4852-bbca-84066305491d">

Reply:
What?! Oh.... man.... I used patch recently and didn't even remember I did.... I am getting so bad in memory.... so sorry, if I remembered I should point to you this file....
I approve... but thank you for reminding me!!